File: backend/app/schemas/complaint.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from app.schemas.investigation import SignalResponse

class ComplaintCreate(BaseModel):
    investigation_id: str
    title: str
    category: str
    description: str
    included_signal_ids: Optional[List[str]] = []

class ComplaintUpdate(BaseModel):
    title: Optional[str] = None
    category: Optional[str] = None
    description: Optional[str] = None
    included_signal_ids: Optional[List[str]] = None

class ComplaintConfirm(BaseModel):
    user_confirmed: bool

class ComplaintEvidenceItem(BaseModel):
    id: str
    signal_id: str
    included: bool
    signal: Optional[SignalResponse] = None

    class Config:
        from_attributes = True

class ComplaintResponse(BaseModel):
    id: str
    complaint_identifier: str
    user_id: str
    investigation_id: str
    title: str
    category: str
    description: str
    status: str
    user_confirmed: bool
    created_at: datetime

    class Config:
        from_attributes = True

class ComplaintDetailResponse(BaseModel):
    id: str
    complaint_identifier: str
    user_id: str
    investigation_id: str
    title: str
    category: str
    description: str
    status: str
    user_confirmed: bool
    created_at: datetime
    
    # Target investigation context
    domain: str
    normalized_url: str
    platform: str
    risk_score: int
    risk_level: str
    
    # Evidence list
    evidence_items: List[ComplaintEvidenceItem] = []
    
    # Text declarations
    declaration: str

    class Config:
        from_attributes = True

File: frontend/src/types/complaint.ts
import { AnalysisSignal } from './investigation';

export interface ComplaintEvidenceItem {
  id: string;
  signal_id: string;
  included: boolean;
  signal?: AnalysisSignal;
}

export interface ComplaintResponse {
  id: string;
  complaint_identifier: string;
  user_id: string;
  investigation_id: string;
  title: string;
  category: string;
  description: string;
  status: 'DRAFT' | 'READY' | 'SUBMITTED' | 'CLOSED';
  user_confirmed: boolean;
  created_at: string;
}

export interface ComplaintDetailResponse extends ComplaintResponse {
  domain: string;
  normalized_url: string;
  platform: string;
  risk_score: number;
  risk_level: string;
  evidence_items: ComplaintEvidenceItem[];
  declaration: string;
}

export interface ComplaintCreate {
  investigation_id: string;
  title: string;
  category: string;
  description: string;
  included_signal_ids?: string[];
}

export interface ComplaintUpdate {
  title?: string;
  category?: string;
  description?: string;
  included_signal_ids?: string[];
}
